Secure-Software-Development-Policy/Zowe-SSDP-Organization/Policy.md

```diff
@@ -0,0 +1,206 @@
+
+```TODO: consider removing or mergin with the previous one or be flexible enough to have an option for different kinds of contributors :individual, squad, company with own tool chains?```
+
+- A2.3 Generate Evidence and Artifacts (#ID: ZSSD-OP:PO-ST-GEA)\
+```#REF: SSDF:PO.3.3```\
+```#KEEP```
+
+Configure tools to generate evidence and artifacts of their support of secure software development practices as defined by Zowe.
+
+#### A3. Security Criteria (#ID: ZSSD-OP:PO-SC) 
+```#REF:  SSFD:PO.4 Define and Use Criteria for Software Security Checks)```\
+```#KEEP```
+
+Provide some kind of automatic validation for the Zowe Secure Development criteria - validate during pipeline execution.
+
+```#Q: Should we require projects to provide own checks for the criteria they declare?```\
+```How to ensure, enforce?```\
+```Code level security checks.```\
+```Dependency level security checks.```\
+```SonarCloud is already used and has some security checks.```\
+
+```TODO: Rephrase - maintain some criteria at Zowe level, while allowing projects to have additional specific criteria.```
+
+- A3.1 Software Security Checks (#ID: ZSSD-OP:PO-SC-SSC)
+```#REF: SSDF:PO.4.1```
+```#KEEP```  
+
+Define criteria for software security checks and track them throughout the SDLC.
+
+- A3.2 Safeguard the Necessary Information (#ID: ZSSD-OP:PO-SC-SNI):
+``` #REF: SSDF:PO.4.2```\
+```#KEEP```
+```#TOO-BROAD```
+
+Implement processes, mechanisms, etc. to gather and safeguard the necessary information in support of the criteria.
+
+#### A4. Secure Environments (#ID: ZSSD-OP:PO-SE)
+```#REF: SSFD:PO.5 Implement and Maintain Secure Environments for Software Development```
+
+- A4.1 Separate and Protect the Environments (#ID: ZSSD-OP:PO-SE-SPE):\
+```#REF: SSDF:PO.5.1```
+
+Separate and protect each environment involved in software development.
+
+```#Q: What is Dev and Test in the Zowe understanding? Are the pipelines on separate environments?```\  
+```#Q: Do we have a Prod at all?```\
+```#Q Do we mix environments in any sense?```s
+
+
+- A4.2 Harden Development Endpoints (#ID: ZSSD-OP:PO-SE-HDE)\
+```#REF: SSDF:PO.5.2```
+
+Secure and harden development endpoints (i.e., endpoints for software designers, developers, testers, builders, etc.\) to perform development-related tasks using a risk-based approach.
+
+```#Q: Can we require and enforce that? From organizations like IBM, BC and Rocket probably yes, but individuals and smaller teams may just not have resources or the will for that.```  
+
+
+### B. Protect the Software (#ID: ZSSD-OP:PS)
+```#REF: SSFD:PS Protect the Software```
+
+#### B1. Protect All Code (#ID: ZSSD-OP:PS-PAC)
+
+```#REF: SSDF:PS.1 Protect All Forms of Code from Unauthorized Access and Tampering```
+
+- B1.1 Store All Code (#ID: ZSSD-OP:PS-PAC-SAC)
+
+Store all forms of code, including source code and executable code, based on the principle of least privilege so that only authorized personnel, tools,services, etc. have the necessary forms of access.
```

Comment:
Also, store the source code on github. furthermore, store it on github.com/zowe because the OMP/LF has some requirements on needing access to it.
pipeline code may be elsewhere as needed (formerly jenkins)